.travis.yml

```diff
@@ -41,11 +47,29 @@ cache:
 notifications:
   email: false
 
-# 5. Run maven install before running lint-java.
+# 5. Run maven install before running lint.
 install:
   - export MAVEN_SKIP_RC=1
   - build/mvn -T 4 -q -DskipTests -Pmesos -Pyarn -Phadoop-2.3 -Pkubernetes -Pkinesis-asl -Phive -Phive-thriftserver install
 
-# 6. Run lint-java.
+# 6. Run test
 script:
-  - dev/lint-java
+  - if [[ $TEST_TO_RUN == "lint-java" ]] ; then
+      dev/lint-java;
+    else
+      echo "SKIPPED";
+    fi
+  - if [[ $TEST_TO_RUN == "lint-scala" ]] ; then
+      dev/lint-scala;
```

Comment:
Travis munges that whole block onto one line during execution, which would cause the problem for the shell command.

Something I had tried on my test repo (here the lint-java has the semi-colons and lint-scala doesn't):

0.00s$ if [[ $TEST_TO_RUN == "lint-java" ]] ; then dev/lint-java; else echo "SKIPPED"; fi
SKIPPED
**The command "if [[ $TEST_TO_RUN == "lint-java" ]] ; then dev/lint-java; else echo "SKIPPED"; fi" exited with 0.**


0.00s$ if [[ $TEST_TO_RUN == "lint-scala" ]] ; then dev/lint-scala else echo "SKIPPED" fi
/home/travis/build.sh: eval: line 60: syntax error: unexpected end of file
**The command "if [[ $TEST_TO_RUN == "lint-scala" ]] ; then dev/lint-scala else echo "SKIPPED" fi" exited with 1.**